engine/runtime-test-instruments/src/main/java/org/enso/interpreter/test/instruments/tck/EnsoTckLanguageProvider.java

```diff
@@ -0,0 +1,86 @@
+package org.enso.interpreter.test.instruments.tck;
+
+import java.util.Collection;
+import java.util.List;
+
+import org.graalvm.polyglot.Context;
+import org.graalvm.polyglot.Source;
+import org.graalvm.polyglot.Value;
+import org.graalvm.polyglot.tck.LanguageProvider;
+import org.graalvm.polyglot.tck.Snippet;
+import org.graalvm.polyglot.tck.TypeDescriptor;
+
+public class EnsoTckLanguageProvider implements LanguageProvider {
+  public EnsoTckLanguageProvider() {
+  }
+
+  @Override
+  public String getId() {
+    return "enso";
+  }
+
+  @Override
+  public Value createIdentityFunction(Context context) {
+    var id = context.eval("enso", """
+    id a = a
+    """).invokeMember("eval_expression", "id");
+    return id;
+  }
+
+  @Override
+  public Collection<? extends Snippet> createValueConstructors(Context context) {
+    return List.of();
+  }
+
+  @Override
+  public Collection<? extends Snippet> createExpressions(Context context) {
+    var plus = context.eval("enso", """
+    plus a b = a + b
+    """).invokeMember("eval_expression", "plus");
+
+    return List.of(
```

Comment:
Ideally we want to expand these values, expressions, constructors, etc. to cover other Enso types - not just `NUMBER` and `STRING`.